Source/Surfaces/SMTopographyClass.f90

```diff
@@ -127,6 +132,43 @@ FUNCTION heightAtTopography(self,t)  RESULT(x)
      END FUNCTION heightAtTopography
 !
 !//////////////////////////////////////////////////////////////////////// 
+! 
+     REAL(KIND=RP) FUNCTION gaussianCurvatureBaseAt(self, t)  
+        IMPLICIT NONE  
+        CLASS(SMTopography) :: self
+        REAL(KIND=RP)       :: t(2)
+        REAL(KIND=RP)       :: gradF(2)
+        REAL(KIND=RP)       :: xp(3)  , xm(3)  , yp(3)  , ym(3)  , x(3)
+        REAL(KIND=RP)       :: xpyp(3), xmyp(3), xpym(3), xmym(3)
+        REAL(KIND=RP)       :: hessian(2,2)
+      
+        x  = self % heightAt(t)
+        xp = self % heightAt(t + [gc_dx,0.0_RP])
+        xm = self % heightAt(t - [gc_dx,0.0_RP])
+        yp = self % heightAt(t + [0.0_RP,gc_dx])
+        ym = self % heightAt(t - [0.0_RP,gc_dx])
+      
+        gradF(1) = 0.5_RP*(xp(3) - xm(3))/gc_dx
+        gradF(2) = 0.5_RP*(yp(3) - ym(3))/gc_dx
+         
+        hessian(1,1) = (xp(3) - 2.0_RP*x(3) + xm(3))/gc_dx2
+        hessian(2,2) = (yp(3) - 2.0_RP*x(3) + ym(3))/gc_dx2
+        
+        
+        xpyp = self % heightAt(t + [gc_dx,gc_dx])
+        xmyp = self % heightAt(t + [-gc_dx,gc_dx])
+        xmym = self % heightAt(t - [gc_dx,gc_dx])
+        xpym = self % heightAt(t + [gc_dx,-gc_dx])
+        hessian(1,2) = 0.25_RP*(xpyp(3) - xmyp(3) - xpym(3) + xmym(3))/gc_dx2
+        hessian(2,1) = hessian(1,2)
+        
+        gaussianCurvatureBaseAt = (hessian(1,1)*hessian(2,2) - hessian(1,2)*hessian(2,1)) &
+                                  /(1.0_RP + gradF(1)**2 + gradF(2)**2)**2
```

Comment:
Thinking back, I may have not used the GaussianCurvature(f, x, y) to more efficiently compute the hessian through re-use, but I'd have to check to make sure. Efficiency is key, because this is really expensive. See final comment below.

Reply:
This has now been changed to call the GC function

Reply:
Great! I know that the Gaussian curvature computations are very expensive and might need some optimization and/or thought about the code structure. Thinking about this, having a mesh generator that can properly size and extrude complex bottom data is an interesting / unique feature (as far as I know). @fluidnumerics-joe Would this be a feature to highlight HOHQMesh and how it can be used in ocean modelling? I am not as familiar with this community